src/cocotb/share/lib/embed/gpi_embed.cpp

```diff
@@ -139,16 +82,41 @@ extern "C" COCOTB_EXPORT void _embed_init_python(void) {
         }
     }
 
-    to_python();
-    // must set program name to Python executable before initialization, so
-    // initialization can determine path from executable
-    set_program_name_in_venv();
-    Py_Initialize(); /* Initialize the interpreter */
-    PySys_SetArgvEx(1, argv, 0);
-
-    /* Swap out and return current thread state and release the GIL */
-    gtstate = PyEval_SaveThread();
-    to_simulator();
+    const char *python_bin_path = getenv("PYGPI_PYTHON_BIN");
+    if (!python_bin_path) {
+        LOG_ERROR(
+            "PYGPI_PYTHON_BIN variable not set, can't initialize Python.");
+        return;
+    }
+    LOG_INFO("Using Python interpreter at %s", python_bin_path);
+
+    // Intialize the Python initialization Config object.
+    PyConfig config;
+    PyConfig_InitPythonConfig(&config);
+    DEFER(PyConfig_Clear(&config));
+
+    // Set program name to the Python interpreter at PYGPI_PYTHON_BIN.
+    // This is necessary to set up sys.path as if the PyGPI user was invoked as
+    // a Python script.
+    {
+        auto status = PyConfig_SetBytesString(&config, &config.program_name,
+                                              python_bin_path);
```

Comment:
This implicitly PreInitializes the Python interpreter and uses DecodeLocale under the hood, so we don't need that bit of code anymore.